modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java

```
@@ -920,162 +922,170 @@ protected ServiceProvider generateServiceProviderWithRoles(OAuthConsumerAppDTO a
+        JSONObject json = (JSONObject) parser.parse(EntityUtils.toString(response.getEntity()));
+        return ((JSONArray) ((JSONObject) ((JSONArray) json.get("keys")).get(0)).get("x5c")).get(0).toString();
+    }
 
-	public String getRoleV2ResourceId(String roleName, String audienceType, String OrganizationId) throws Exception {
+    /**
+     * Authorize list of SYSTEM APIs to an application.
+     *
+     * @param applicationId  Application id.
+     * @param apiIdentifiers API identifiers to authorize.
+     * @throws Exception Error occured while authorizing APIs.
+     */
+    public void authorizeSystemAPIs(String applicationId, List<String> apiIdentifiers) throws Exception {
+
+        apiIdentifiers.stream().forEach(apiIdentifier -> {
+            try {
+                List<APIResourceListItem> filteredAPIResource =
+                        restClient.getAPIResourcesWithFiltering("identifier+eq+" + apiIdentifier);
+                if (filteredAPIResource == null || filteredAPIResource.isEmpty()) {
+                    return;
+                }
+                String apiId = filteredAPIResource.get(0).getId();
+                // Get API scopes.
+                List<ScopeGetModel> apiResourceScopes = restClient.getAPIResourceScopes(apiId);
+                AuthorizedAPICreationModel authorizedAPICreationModel = new AuthorizedAPICreationModel();
+                authorizedAPICreationModel.setId(apiId);
+                authorizedAPICreationModel.setPolicyIdentifier("RBAC");
+                apiResourceScopes.forEach(scope -> {
+                    authorizedAPICreationModel.addScopesItem(scope.getName());
+                });
+                restClient.addAPIAuthorizationToApplication(applicationId, authorizedAPICreationModel);
+            } catch (Exception e) {
+                throw new RuntimeException("Error while authorizing system API " + apiIdentifier + " to application "
+                        + applicationId, e);
+            }
```

You have already thrown a catched exception from the method. Given that, in this try catch block catch the exact exception and you can throw an Exception wrapping that than the runtime exception

---

Or else remove 'throw exception' and have the runtime exception, so it will be thrown or logged eventually for the respective test method

---

Fixed with the commit